src/seq/bioseq.jl

```diff
@@ -1100,3 +1098,130 @@ function seqmatrix{A<:Alphabet}(vseq::AbstractVector{BioSequence{A}}, major::Sym
+as julia arrays are laid out in column major order.
+
+If the parameter major is set to `:seq`, the matrix is created such that each
+sequence is placed in one column i.e. the matrix is laid out in sequence-major
+order.
+This means that iteration over each sequence is efficient,
+as julia arrays are laid out in column major order.
+
+# Examples
+```julia
+julia> seqs = [dna"AAA", dna"TTT", dna"CCC", dna"GGG"]
+4-element Array{Bio.Seq.BioSequence{Bio.Seq.DNAAlphabet{4}},1}:
+ 3nt DNA Sequence:
+AAA
+ 3nt DNA Sequence:
+TTT
+ 3nt DNA Sequence:
+CCC
+ 3nt DNA Sequence:
+GGG
+
+julia> seqmatrix(seqs, :site, UInt8)
+4×3 Array{UInt8,2}:
+ 0x01  0x01  0x01
+ 0x08  0x08  0x08
+ 0x02  0x02  0x02
+ 0x04  0x04  0x04
+
+julia> seqmatrix(seqs, :seq, UInt8)
+3×4 Array{UInt8,2}:
+ 0x01  0x08  0x02  0x04
+ 0x01  0x08  0x02  0x04
+ 0x01  0x08  0x02  0x04
+```
+"""
+function seqmatrix{A<:Alphabet,T}(vseq::AbstractVector{BioSequence{A}}, major::Symbol, conv::Type{T})
+    nsites = length(vseq[1])
+    nseqs = length(vseq)
+    @inbounds for i in 2:nseqs
+        length(vseq[i]) == nsites || throw(ArgumentError("Sequences in vseq must be of same length"))
+    end
+    if major == :site
+        mat = Matrix{T}(nseqs, nsites)
+        @inbounds for seq in 1:nseqs, site in 1:nsites
+            mat[seq, site] = convert(T, vseq[seq][site])
+        end
+        return mat
+    elseif major == :seq
+        mat = Matrix{T}(nsites, nseqs)
+        @inbounds for seq in 1:nseqs, site in 1:nsites
+            mat[site, seq] = convert(T, vseq[seq][site])
+        end
+        return mat
+    else
+        throw(ArgumentError("major must be :site or :seq"))
+    end
+end
+
+# Consensus
+# ---------
+
+"""
+    majorityvote(seqs::AbstractVector{DNASequence})
+
+Construct a sequence that is a consensus of a vector of sequences.
+
+The consensus is established by a simple majority vote rule, where amiguous
+nucleotides cast an equal vote for each of their possible states.
+For each site a winner(s) out of A, T(U), C, or G is determined, in the cases
+of ties the ambiguity symbol that unifies all the winners is returned.
+E.g if A and T tie, then W is inserted in the consensus. If all A, T, C, and G
+tie at a site, then N is inserted in the consensus.
+
+# Examples
+```julia
+julia> seqs = [dna"CTCGATCGATCC", dna"CTCGAAAAATCA", dna"ATCGAAAAATCG", dna"ATCGGGGGATCG"]
+
+4-element Array{Bio.Seq.BioSequence{Bio.Seq.DNAAlphabet{4}},1}:
+ CTCGATCGATCC
+ CTCGAAAAATCA
+ ATCGAAAAATCG
+ ATCGGGGGATCG
+
+julia> majorityvote(seqs)
+12nt DNA Sequence:
+MTCGAAARATCG
+```
+"""
+function majorityvote{A<:NucleotideAlphabets}(seqs::AbstractVector{BioSequence{A}})
+    mat = seqmatrix(seqs, :site, UInt8)
+    nsites = size(mat, 2)
+    nseqs = size(mat, 1)
+    result = BioSequence{A}(nsites)
+    @inbounds for site in 1:nsites
+        votes = zeros(Int, 16)
+        for seq in 1:nseqs
+            nuc = mat[seq, site]
+            votes[1] += nuc == 0x00
```

Comment:
How should gap symbols be handled? If the gap symbol does no vote, I believe this line is not necessary.